packages/file-search/src/common/file-search-service.ts

```diff
@@ -33,13 +33,24 @@ export interface FileSearchService {
 
 export const FileSearchService = Symbol('FileSearchService');
 export namespace FileSearchService {
-    export interface Options {
-        rootUris: string[]
+    export interface BaseOptions {
+        useGitIgnore?: boolean
+        includePatterns?: string[]
+        excludePatterns?: string[]
+    }
+    export interface Roots {
+        [rootUri: string]: BaseOptions
+    }
+    export interface Options extends BaseOptions {
+        rootUris?: string[]
+        roots?: Roots
```

Comment:
does this mean "option per root" ? if so, could we rename it into something more descriptive ? i have no idea what it could be renamed into as I am bad at naming things :)

Reply:
`rootOptions`?

Reply:
It's confusing as well. It means options for a root, not per a root. Do you have better suggestion or we stick with `roots`? To me is fine, since each property is a root with an associated options.

Reply:
how does "optionPerRoot" or "perRootOption" sound?

Reply:
i would rename it to `rootOptions` later, `optionPerRoot` looks very atypical

Reply:
@elaihau I've renamed to `rootOptions`, good to merge now?

Reply:
yes for sure. the pr approval is still in place :)